<?xml version="1.0" ?>

<robot name="end_effector_simple">
  <!--
    Frames are defined such that they all align with the base frame when all
    joints are at position zero.

    In zero-configuration the origins of all joints are in one line.

    In zero-configuration the origin of the base link is exactly above the
    finger tip which should make it easy to place the finger in the world.
    -->
  <!-- Define the global base_link and place all other objects relative to it. -->
  
  <material name="finger_0_material">
    <color rgba="0.6 0.0 0.0 1.0"/>
  </material>

  <link name="base_link" />

  <link name="end_effector_simple">
   <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.0195"/>
      </geometry>
      <material name="finger_0_material"/>
    </visual>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="1"/>
    </inertial>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.0195"/>
      </geometry>
                  <drake:proximity_properties>
        <drake:mu_static value="1"/>
        <drake:mu_dynamic value="1"/>
      </drake:proximity_properties>
    </collision>
  </link>
  
  <link name="end_effector_simple_fake_x" />
  
  <link name="end_effector_simple_fake_y" />
  
  <joint name="end_effector_simple_to_base_x" type="prismatic">
    <parent link="base_link"/>
    <child link="end_effector_simple_fake_x"/>
    <axis xyz="1 0 0"/>
  </joint>
  <joint name="end_effector_simple_to_base_y" type="prismatic">
    <parent link="end_effector_simple_fake_x"/>
    <child link="end_effector_simple_fake_y"/>
    <axis xyz="0 1 0"/>
  </joint>
    <joint name="end_effector_simple_to_base_z" type="prismatic">
    <parent link="end_effector_simple_fake_y"/>
    <child link="end_effector_simple"/>
    <axis xyz="0 0 1"/>
  </joint>
  
   <transmission name="end_effector_simple_to_base_trans_x" type="SimpleTransmission">
    <actuator name="end_effector_simple_to_base_act_x"/>
    <joint name="end_effector_simple_to_base_x"/>
    <mechanicalReduction>1</mechanicalReduction>
  </transmission>
  
     <transmission name="end_effector_simple_to_base_trans_y" type="SimpleTransmission">
    <actuator name="end_effector_simple_to_base_act_y"/>
    <joint name="end_effector_simple_to_base_y"/>
    <mechanicalReduction>1</mechanicalReduction>
  </transmission>
  
     <transmission name="end_effector_simple_to_base_trans_z" type="SimpleTransmission">
    <actuator name="end_effector_simple_to_base_act_z"/>
    <joint name="end_effector_simple_to_base_z"/>
    <mechanicalReduction>1</mechanicalReduction>
  </transmission>
  
  <drake:collision_filter_group name="finger_0_group">
  <drake:member link="end_effector_simple"/>
  <drake:ignored_collision_filter_group name="finger_0_group"/>
  </drake:collision_filter_group>
</robot>